### PR TITLE
DOC-6415 DSL deprecation

### PR DESCRIPTION
**Pull request by RichardSmedley**

### CODE REVIEW

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc [hunk 1]
```diff
@@ -510,6 +510,15 @@ Exception in thread "main" com.couchbase.client.core.error.ParsingFailedExceptio
 
 Not only does it throw a `CouchbaseException`, it also tries to map it to a specific exception type and include extensive contextual information for a better troubleshooting experience.
 
+==== DSL Deprecation
+
+The fluent API feature tried out in SDK 2 was found not to scale well for larger queries, nor for SQL++.
```
Comment (ingenthr) [inline modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc]: "tried out" makes it sound like it was experimental, which it wasn't.  I'd just recommend removing those words.

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc [hunk 2]
```diff
@@ -510,6 +510,15 @@ Exception in thread "main" com.couchbase.client.core.error.ParsingFailedExceptio
...
+There was also some impedance mismatch with the Java language.
```
Comment (ingenthr) [inline modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc]: while true, this doesn't get better by manipulating strings.  Just remove it I'd suggest..

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc [hunk 3]
```diff
@@ -510,6 +510,15 @@ Exception in thread "main" com.couchbase.client.core.error.ParsingFailedExceptio
...
+For these reasons it is not amongst the features initially carried over to SDK 3.0.
```
Comment (ingenthr) [inline modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc]: -"initially"

**Review by ingenthr — CHANGES_REQUESTED** [review]

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc (hunk 1 above)
Comment (daschl) [reply modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc]: +1

File: modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc (hunk 2 above)
Comment (daschl) [reply modules/project-docs/pages/migrating-sdk-code-to-3.n.adoc]: +1

**Review by daschl — CHANGES_REQUESTED** [review]

**Review by daschl — APPROVED** [review]

**Pull request merged**